src/compiler.ts

```diff
@@ -725,7 +725,16 @@ export class Compiler extends DiagnosticEmitter {
     for (let i = 0, k = functionTable.length; i < k; ++i) {
       functionTableNames[i] = functionTable[i].internalName;
     }
-    module.addFunctionTable("0", tableBase + functionTable.length, Module.UNLIMITED_TABLE, functionTableNames, module.i32(tableBase));
+
+    var tableSize = tableBase + functionTable.length;
+    module.addFunctionTable(
+      "0",
+      tableSize,
+      // use fixed size for non-imported tables
+      options.importTable ? Module.UNLIMITED_TABLE : tableSize,
```

Comment:
Is this always desirable when the table is exported? Seems that `table.grow` could be called on it externally for some reason. Eventually, we might also want to call `table.grow` internally, which we can intercept, but the external one is a little tricky.

Reply:
I can prohibit the same for exported tables

Reply:
But table.grow will not work. Perhaps we should add a flag that allows you to expand the table? This is a very rare case anyway

Reply:
Good thing is that `--exportTable` is not enabled by default, so depending on whether the option is given, it always does what's least surprising.